android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ContactsModule.java

```diff
@@ -61,116 +108,127 @@ public void getContactsAsync(final ReadableMap options, final Promise promise) {
+        FULL_PROJECTION.add(CommonDataKinds.StructuredName.PHONETIC_FAMILY_NAME);
+      }
 
-            HashMap<String, String> organization = getOrganizationFromContentResolver(id, cr);
-            if (organization != null) {
-              if (fieldsSet.contains("company")) {
-                company = organization.get("company");
-              }
-              if (fieldsSet.contains("jobTitle")) {
-                jobTitle = organization.get("jobTitle");
-              }
-            }
+      if (fieldsSet.contains("phoneticMiddleName")) {
+        FULL_PROJECTION.add(CommonDataKinds.StructuredName.PHONETIC_MIDDLE_NAME);
+      }
 
-            HashMap<String, WritableArray> collections = new HashMap<>();
-            collections.put("emails", fieldsSet.contains("emails") ?
-                getEmailsFromContentResolver(id, cr) : null);
-            collections.put("phoneNumbers", fieldsSet.contains("phoneNumbers") ?
-                getPhoneNumbersFromContentResolver(id, cr) : null);
-            collections.put("addresses", fieldsSet.contains("addresses") ?
-                getAddressesFromContentResolver(id, cr) : null);
-            collections.put("instantMessageAddresses", fieldsSet.contains("instantMessageAddresses") ?
-                getInstantMessageAddressesFromContentResolver(id, cr) : null);
-            collections.put("urlAddresses", fieldsSet.contains("urlAddresses") ?
-                getUrlAddressesFromContentResolver(id, cr) : null);
-            collections.put("relationships", fieldsSet.contains("relationships") ?
-                getRelationshipsFromContentResolver(id, cr) : null);
-
-            for (String fieldName : collections.keySet()) {
-              WritableArray value = collections.get(fieldName);
-              if (value != null && value.size() > 0) {
-                contact.putArray(fieldName, value);
-              }
-            }
+      if (fieldsSet.contains("namePrefix")) {
+        FULL_PROJECTION.add(CommonDataKinds.StructuredName.PREFIX);
+      }
 
-            if (note != null && !note.isEmpty()) {
-              contact.putString("note", note);
-            }
-            if (company != null && !company.isEmpty()) {
-              contact.putString("company", company);
-            }
-            if (jobTitle != null && !jobTitle.isEmpty()) {
-              contact.putString("jobTitle", jobTitle);
-            }
+      if (fieldsSet.contains("nameSuffix")) {
+        FULL_PROJECTION.add(CommonDataKinds.StructuredName.SUFFIX);
+      }
 
-            if (fetchSingleContact) {
-              promise.resolve(contact);
+      cursor = cr.query(
+          ContactsContract.Data.CONTENT_URI,
+          FULL_PROJECTION.toArray(new String[FULL_PROJECTION.size()]),
+          selection,
+          selectionArgs.toArray(new String[selectionArgs.size()]),
+          null);
+    }
+    if (cursor != null) {
+      try {
+        // populate list of contacts from cursor query
```

Comment:
Some comments are probably not quite necessary - you can leave the ones that describe something tricky.

Reply:
check